### PR TITLE
ADD yolov4 in models.yml

### PR DESCRIPTION
**Pull request by cabelo**
### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [ ] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Comment by dkurt** [comment]
There is already `yolo` family there, so the following command should be used for the samples:
```
python3 object_detection.py yolo --model yolov4.weights --config yolov4.cfg
```

**Comment by cabelo** [comment]
> There is already `yolo` family there, so the following command should be used for the samples:
> 
> ```
> python3 object_detection.py yolo --model yolov4.weights --config yolov4.cfg
> ```

We can exchange yolov3.weights for yolov4.weights in the model.yml file.

What do you think?

File: samples/dnn/models.yml [hunk 1]
```diff
@@ -15,11 +15,11 @@ opencv_fd:
   rgb: false
   sample: "object_detection"
 
-# YOLO object detection family from Darknet (https://pjreddie.com/darknet/yolo/)
-# Might be used for all YOLOv2, TinyYolov2 and YOLOv3
+# YOLO object detection family from Darknet (https://github.com/AlexeyAB/darknet#pre-trained-models)
```
Comment (dkurt) [inline samples/dnn/models.yml:18]: Please refer to origin repository as well:
```
# YOLO object detection family from Darknet. Might be used for
#   * YOLOv2, TinyYolov2 and YOLOv3 (https://pjreddie.com/darknet/yolo/)
#   * YOLOv4 (https://github.com/AlexeyAB/darknet)
```

File: samples/dnn/models.yml [hunk 2]
```diff
@@ -15,11 +15,11 @@ opencv_fd:
...
+# Might be used for all YOLOv2, TinyYolov2, YOLOv3 and YOLOv4
 yolo:
-  model: "yolov3.weights"
-  config: "yolov3.cfg"
+  model: "yolov4.weights"
+  config: "yolov4.cfg"
```
Comment (dkurt) [inline samples/dnn/models.yml:22]: Please keep an origin behavior.

**Comment by dkurt** [comment]
Please target this PR to 3.4 branch. Do locally:
```
git checkout 3.4
git br -m yolov4-in-model tmp
git checkout -b yolov4-in-model
git merge --squash tmp
git commit -m "ADD yolov4 in models.yml"
```

File: samples/dnn/models.yml (hunk 1 above)
Comment (cabelo) [reply samples/dnn/models.yml:18]: ok

File: samples/dnn/models.yml (hunk 2 above)
Comment (cabelo) [reply samples/dnn/models.yml:22]: ok

**Pull request closed without merging**